addons/assignGear/functions/fnc_addSupplyBoxActions.sqf

```diff
@@ -0,0 +1,94 @@
+#include "script_component.hpp"
+/*
+ * Author: Lambda.Tiger
+ * Adds actions to allow a client to mark a supply box using smoke, glow sticks
+ * and or map markers. Smoke and glow sticks are enabled with a marking level
+ * of 1 or greater, and map markers at level 2 or greater.
+ *
+ * Arguments:
+ * 0: Box <OBJECT>
+ * 1: Marking level <SCALAR>
+ *    >=1 allows smoke and chem lights to be added
+ *    >=2 allows for a map marker to be placed/updated
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorObject, 2] call potato_assignGear_fnc_addSupplyBoxActions.sqf
+ *
+ * Public: No
+ */
+#define LARGE_UNIQUE_MARKER_INDEX 934648
+params ["_theBox", "_markingLevel"];
+
+if (_markingLevel < 1 || !hasInterface) exitWith {};
+
+private _action = [
+    "markBoxBase",
+    "Mark Box",
+    "\a3\ui_f\data\igui\cfg\simpletasks\types\move_ca.paa", {
+        private _marker = _target getVariable [QGVAR(boxMarker), ""];
+        if (getMarkerPos _marker isNotEqualTo [0, 0, 0]) then {
+            _marker setMarkerPos (getPosATL _target);
+        };
+    },
+    {true}
+] call ACEFUNC(interact_menu,createAction);
+[_theBox, 0, ["ACE_MainActions"], _action] call ACEFUNC(interact_menu,addActionToObject);
+
+_action = [
+    "markBoxSmoke",
+    "Smoke",
+    "\a3\Modules_F_Curator\Data\portraitSmoke_ca.paa", {
+    private _smoke = createVehicle ["SmokeShellYellow", ASLToAGL getPosASL _target, [], 0, "CAN_COLLIDE"];
+    _smoke attachTo [_target, [0, 0, 0]];
+    _target setVariable [QGVAR(smokeAvailable), false];
+    },
+    {_target getVariable [QGVAR(smokeAvailable), true]}
```

Comment:
Q: only can do once?

Reply:
Yeah, we could make it more than once, but also that means logi can just use the box as mobile cover & concealment .

Reply:
just once is good!

Reply:
i’m fine with it just once, i just wanted to make sure i didn’t miss it reactivating

Reply:
@kil0byt3, thoughts?